packages/core/src/components/ImageOverlay.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react'
+import { ExamComponentProps } from '../createRenderChildNodes'
+import { mapChildElements, mapChildNodes } from '../dom-utils'
+import { faEye } from '@fortawesome/free-solid-svg-icons/faEye'
+import { faEyeSlash } from '@fortawesome/free-solid-svg-icons/faEyeSlash'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import ImageOverlayImage from './ImageOverlayImage'
+
+function ImageOverlay({ element, renderChildNodes }: ExamComponentProps) {
+  const sliders = mapChildElements(element, (childElement, index) => {
+    const label = mapChildNodes(childElement, (childNode) => {
+      return childNode.textContent
+    })
+    return {
+      label: label.length > 0 ? label[0] : null,
+      opacity: index === 0 ? 100 : 0,
+      element: childElement,
+    }
+  })
+  return <Sliders sliders={sliders} element={element} renderChildNodes={renderChildNodes} />
+}
+
+interface SlidersProps {
+  sliders: SliderProps[]
+}
+
+function Sliders({ sliders, renderChildNodes }: SlidersProps & ExamComponentProps) {
+  const [sliderValues, setSliderValue] = useState(sliders)
+
+  const imagesContainerWidth = sliderValues[0]?.element?.getAttribute('width') || null
+
+  const onSetSliderValue = (event: React.ChangeEvent<HTMLInputElement>, index: number) => {
+    const updatedSliders = [...sliders]
+    updatedSliders[index].opacity = parseInt(event.target.value)
+    setSliderValue(updatedSliders)
+  }
+
+  return (
+    <div className="e-mrg-b-1" style={{ maxWidth: imagesContainerWidth ? `${imagesContainerWidth}px` : '100%' }}>
+      {sliderValues.map((slider, index) => (
+        <Slider
+          key={index}
```

Comment:
Tuli mieleen että kun käytetään näitä indeksejä key-arvona niin siinä on muistaakseni aina pieni bugin riski, että kun poistaa jonkun noden listan välistä niin muutos ei välttämättä näy oikein, koska keyt näyttää Reactin suuntaan samalta kuin aikaisemminkin. Mutta jos tämä toimii niin ei ongelmaa!

Reply:
En kyllä tästä kantaisi huolta, kun slider-lista ei ole muokattavissa. Pikaisella haulla näkyy exam-enginessä olevan muutamassa muussakin kohdassa käytetty indeksi samaan tarkoitukseen, varmaan samalla logiikalla.